### PR TITLE
Add support for request tracing via X-Request-ID header

### PR DESCRIPTION
**Pull request by itskingori**
If the `X-Request-ID` is passed along, then it's easier to follow the chain of calls upstream. See commits for details. Each commit is an idea.

### CODE REVIEW

File: config/log.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 log_format main '$remote_addr - $remote_user [$time_local]  $status '
                 '"$request" $body_bytes_sent "$http_referer" '
-                '"$http_user_agent" "$http_x_forwarded_for"';
+                '"$http_user_agent" "$http_x_forwarded_for" $proxy_x_request_id';
```
Comment (itskingori) [inline config/log.conf:3]: I want to improve this log format to have a little more information but I think that's best done in another PR. Using [the Kubernetes Ingress - Ingress log format](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/log-format/) as inspiration ...

```nginx
log_format upstreaminfo
    '$remote_addr - $remote_user [$time_local] "$request" '
    '$status $body_bytes_sent "$http_referer" "$http_user_agent" '
    '$request_length $request_time [$proxy_upstream_name] [$proxy_alternative_upstream_name] $upstream_addr '
    '$upstream_response_length $upstream_response_time $upstream_status $req_id';
```

File: config/http.conf [hunk 2]
```diff
@@ -39,6 +46,7 @@ http {
   proxy_set_header    X-Forwarded-Proto   $proxy_x_forwarded_proto;
   proxy_set_header    X-Forwarded-Ssl     $proxy_x_forwarded_ssl;
   proxy_set_header    X-Forwarded-Port    $proxy_x_forwarded_port;
+  proxy_set_header    X-Request-ID        $proxy_x_request_id;
```
Comment (itskingori) [inline config/http.conf:49]: Some apps [like Rails with use this as the `request_id` if the `X-Request-ID` exists](https://api.rubyonrails.org/classes/ActionDispatch/RequestId.html). Else, they generate one. So it's certainly useful to set it here because it will be used upstream. See ...

Screenshot 2021-02-10 at 12 01 48

**Review by hadrianvalentine — APPROVED** [review]

**Review by itskingori — COMMENTED** [review]
Some commentary ... 💬

File: config/log.conf (hunk 1 above)
Comment (hadrianvalentine) [reply config/log.conf:3]: I can get behind this 👌

**Review by zacblazic — APPROVED** [review]
Totally beat me to this. 🏆

File: config/log.conf (hunk 1 above)
Comment (itskingori) [reply config/log.conf:3]: > I want to improve this log format to have a little more information but I think that's best done in another PR. Using [the Kubernetes Ingress - Ingress log format](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/log-format/) as inspiration ...

Done in https://github.com/Intellection/docker-nginx-proxy/pull/6.
